### PR TITLE
Budget tracker

### PR DESCRIPTION
**Pull request by anyepeter**
Hello CR😁👋;
In this PR, I built a budget traker app with the following requirements📑

### General requirements
✅ Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
✅ Make sure that you used correct [gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
✅ Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
### Ruby requirements
✅ Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).
### Project requirements
- Design
✅ You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
- Colors.
✅ Typography: font face, size and weight.
✅ Layout: composition and space between elements.
✅ NOTE: In these design guidelines there are several UIs that you won't need for this exercise; also, some pages are not given a design and you will create them following the design guidelines of the other pages.

- Original design idea by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).

 - Interactions
✅ Splash screen
✅ A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
- Sign up and log in pages
✅ The user should be able to register in the app with full name, email and password (all mandatory).
✅ The user can log into the app using email and password.
✅ If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
- Home page (categories page)
✅ When the user logs in, they are presented with the categories page.
✅ For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
✅ When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
✅ There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- Transactions page
✅ For a given category, the list of transactions is presented, ordered by the most recent.
✅ At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
✅ There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
✅ When the user clicks on the "Back" button (<), the user navigates to the home page.
- Add a new category page
✅ The user fills out a form to create a new category, indicating their name and icon (both mandatory).
✅ The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
✅ When the user clicks on the "Back" button (<), the user navigates to the home page.
- Add a new transaction page
✅ The user fills out a form to create a new transaction with:
✅ name (mandatory)
✅ amount (mandatory)
✅ categories (mandatory at least one)
✅ The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
✅ When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+
+I would like to thank Microverse for providing me with the reading material that helped me during this project development.
+
```
Comment (ShadyShawkat) [inline README.md:172]: - [ ] Good job giving credit to the author of the design in the PR description. However, it's a requirement to give appropriate credit to the author of the original design in the readme, as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=The%20Creative%20Commons%20license%20of%20the%20design%20requires%20that%20you%20give%20appropriate%20credit%20to%20the%20author.%20Therefore%2C%20you%20must%20do%20it%20in%20the%20README%20of%20your%20project.) requirement. Kindly do so.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+<!-- TABLE OF CONTENTS -->
+
+# 📗 Table of Contents
+
+- [📖 About the Project](#about-project)
+  - [🛠 Built With](#built-with)
+    - [Tech Stack](#tech-stack)
+    - [Key Features](#key-features)
+  - [🚀 Live Demo](#live-demo)
```
Comment (ShadyShawkat) [inline README.md:14]: - [ ] Please deploy the application on one of the free host services and please make sure that it's working properly, as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=is%20tested%20adequately.-,Deploy%20the%20project,-%2C%20and%20test%20for) requirement

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @anyepeter, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+
+- I would like to thank Microvers for providing this great opportunity and for the project guide as well 👏
+- Huge appriciation to Gregoire Vella for providing design system guidelines since this project is built upon thatopment.
```
Comment (youmari) [inline README.md:175]: - [ ] [ OPTIONAL ] please make sure to correct the typing errors in this sentence like `appriciation and thatopment`

**Review by youmari — APPROVED** [review]
Hi @anyepeter , 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 
## Highlights 
✔️ Clean Design 
✔️ Clean code 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**